typescript/sdk/src/hook/EvmHookModule.ts

```diff
@@ -251,6 +247,48 @@ export class EvmHookModule extends HyperlaneModule<
     return updateTxs;
   }
 
+  /**
+   *  Recursively resolves the HookConfigs as addresses, e.g.
+   *  hook:
+   *     type: aggregationHook
+   *     hooks:
+   *       - "0x7937CB2886f01F38210506491A69B0D107Ea0ad9"
+   *       - beneficiary: "0x865BA5789D82F2D4C5595a3968dad729A8C3daE6"
+   *         maxProtocolFee: "100000000000000000000"
+   *         owner: "0x865BA5789D82F2D4C5595a3968dad729A8C3daE6"
+   *         protocolFee: "50000000000000000"
+   *         type: protocolFee
+   *
+   * This may throw if the Hook address is not a derivable hook (e.g. Custom Hook)
+   */
+  private async resolveHookAddresses(config: HookConfig) {
```

Comment:
As discussed, you might need this @xeno097 for the cli checker. i wonder if it makes sense to put this in the sdk such as the HookReader where we can do all the extensions (remoteRouter, destinationGas)

Reply:
yeah i think having it in the hook reader as a public method would help

Reply:
Moved